packages/design-system/src/scalars/components/fragments/toggle-field/toggle-field.tsx

```diff
@@ -20,11 +23,25 @@ const ToggleField: React.FC<ToggleFieldProps> = ({
   warnings = [],
   required = false,
   className,
-  defaultValue,
+  defaultValue = true,
   description,
+  validateOnBlur,
+  validateOnChange,
 }) => {
   const generatedId = useId();
   const id = idProp ?? generatedId;
+  const [touched, setTouched] = useState(false);
+
+  const validateField = () => {
+    return [
+      ...errors,
+      ...validateRequiredField(value ?? defaultValue, required),
+    ];
+  };
+
+  const errorMerge = validateField();
+  const showError = touched && !!errorMerge.length;
+  const showWarning = touched && !!warnings.length;
```

Comment:
we should reuse this logic in a hook, so we're able to reuse it in each field, otherwise, the same logic will be implemented in each Field component

Reply:
you are right, this will close , because we move to different path for validation